### PR TITLE
Improve validator birthday date

### PR DESCRIPTION
**Pull request by aleeks**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Improve validator birthday date |
| Type? | improvement |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | no |
| How to test? | See below |

### CODE REVIEW

**Comment by PrestaEdit** [comment]
Hi,

While we are here...: maybe we need to return false, here, no ?
https://github.com/aleeks/PrestaShop/blob/7d0f9ffb0bf5e229a85cdbb7113ab7e83fb07060/classes/Validate.php#L524

**Comment by mickaelandrieu** [comment]
Ok, but I'd like to see unit tests, I don't want to see any regression on validators because it's a pain to debug.

You can use a simple [data provider](https://phpunit.de/manual/4.8/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers) and iterate into the samples you have described :)

**Comment by aleeks** [comment]
@mickaelandrieu : ok, I do this tomorrow!

@PrestaEdit : I thought like you at first, but there was a problem with the conditions (missing and returned false..).

` ($birth_date[1] > date('Y') && $birth_date[2] > date('m') && $birth_date[3] > date('d'))` 
=> returned false, and it's ok because it's not a birthdate..

The function name is ambiguous, if we return false to say "yes, it's a birthdate.."
So I think we don't consider the test if is an empty date  or '0000-00-00'.. but why ?..
I will make all tests tomorrow and I will speak with the team to be sure.

**Comment by kpodemski** [comment]
Maybe we could use just `checkdate` function?

**Comment by kpodemski** [comment]
Maybe we could use `checkdate` function + `DateTime` comparison to check wether the date is a future or not? Wouldn't be that simplier?

**Comment by aleeks** [comment]
@kpodemski : yes, +1 for isDate & comparaison. 
What do you think about the first condition and the function return ?

File: tests/Unit/classes/ValidateCoreTest.php [hunk 1]
```diff
@@ -149,6 +157,18 @@ public function isEmailDataProvider()
         );
     }
 
+    public function isBirthDateProvider()
+    {
+        return array(
```
Comment (mickaelandrieu) [inline tests/Unit/classes/ValidateCoreTest.php]: you can also do

```php
    return array(
         'valid_1' => array(true, '1991-04-19'),
    );
```

if you need to add context to each use case

**Comment by maximebiloe** [comment]
Thank you!

**Pull request merged**